internal/cmd/db_create.go

```diff
@@ -165,6 +165,17 @@ var createCmd = &cobra.Command{
 			extensions = "all"
 		}
 
+		var timestamp *time.Time
+		if fromDBFlag != "" {
+			ts, err := time.Parse("2006-01-02T15:04:05", timestampFlag)
+			if err != nil {
+				return fmt.Errorf("provided timestamp was not in 'yyyy-MM-ddThh:mm::ss' format")
+			}
+			timestamp = &ts
+		} else if timestampFlag != "" {
+			return fmt.Errorf("--timestamp cannot be used without specifying --from-db option")
+		}
```

Comment:
⛏️: please extract this into a function that returns (*time.Time, error)`. You can use early returns to avoid nesting conditionals there. 

As @haaawk pointed out, please also handle the `timestampFlag == ""` case.